lib/plugins.js

```diff
@@ -505,12 +505,16 @@ function plugins (settings, deviceId) {
     const tradeEntry = expand(_tradeEntry)
     const execute = tradeEntry.type === 'buy' ? exchange.buy : exchange.sell
 
-    return execute(settings, tradeEntry.cryptoAtoms, tradeEntry.fiatCode, tradeEntry.cryptoCode)
-      .then(() => recordTrade(tradeEntry))
-      .catch(err => {
-        return recordTrade(tradeEntry, err)
-          .then(() => {
-            throw err
+    return recordTrade(tradeEntry)
+      .then(newEntry => {
+        return execute(settings, tradeEntry.cryptoAtoms, tradeEntry.fiatCode, tradeEntry.cryptoCode, newEntry.id)
+          .catch(err => {
+            const data = mergeTradeEntryAndError(tradeEntry, err)
+            const sql = pgp.helpers.update(data, ['error'], 'trades') + ` WHERE id = ${newEntry.id}`
+            return db.none(sql)
+              .then(() => {
+                throw err
+              })
```

Comment:
This is basically adapting the execute function and creating a function that updates the trade with error.

```suggestion
    return recordTrade(tradeEntry)
      .then(newEntry => execute(settings, tradeEntry, newEntry.id)
      .catch(err => {
        updateTradeEntry(tradeEntry, err)
          .catch(console.log)
        throw err
     })
```

Reply:
Not sure if I fully understood this, should I move the update logic to a new function updateTradeEntry? The execute() lost an argument with this changes, is this intentional? If so I'll need to change the buy/sell functions in the exchange.js and where they're called.

Reply:
> Not sure if I fully understood this, should I move the update logic to a new function updateTradeEntry?

right

> The execute() lost an argument with this changes, is this intentional? 

This one was mostly to reduce the params and line length, only thing I did was pass down the entire tradeEntry object instead of the separated fields.